components/Author/Tabs/FeedCard.tsx

```diff
@@ -128,71 +136,32 @@ function FeedCard(props: FeedCardProps) {
     });
   }
 
-  const createVoteHandler = (voteType) => {
-    const voteStrategies = {
-      [UPVOTE]: {
-        increment: 1,
-        getUrl:
-          formattedDocType === "post"
-            ? API.RH_POST_UPVOTE(id)
-            : API.HYPOTHESIS_VOTE({ hypothesisID: id, voteType }),
-      },
-      [DOWNVOTE]: {
-        increment: -1,
-        getUrl:
-          formattedDocType === "post"
-            ? API.RH_POST_DOWNVOTE(id)
-            : API.HYPOTHESIS_VOTE({ hypothesisID: id, voteType }),
-      },
-    };
-
-    const { increment, getUrl } = voteStrategies[voteType];
-
-    const handleVote = async () => {
-      const response = await fetch(getUrl, API.POST_CONFIG()).catch((err) =>
-        emptyFncWithMsg(err)
-      );
-
-      return response;
-    };
-
-    return async (e: SyntheticEvent) => {
-      e.preventDefault();
-      e.stopPropagation();
-
-      if (
-        !isNullOrUndefined(currentUser) &&
-        currentUser.id === created_by.author_profile.id
-      ) {
-        emptyFncWithMsg(
-          `Not logged in or attempted to vote on own ${formattedDocType}.`
-        );
-        return;
-      }
-
-      if (voteState === voteType) {
-        return;
-      } else {
-        setVoteState(voteType);
-        setScore(score + (Boolean(voteState) ? increment * 2 : increment));
-      }
-
-      await handleVote();
-    };
-  };
-
-  async function onPaperVote(voteType) {
-    const curPaper = { ...paper };
-    const increment = voteType === UPVOTE ? 1 : -1;
-    setVoteState(voteType);
-    setScore(score + (Boolean(voteState) ? increment * 2 : increment));
-    voteCallback && voteCallback(index, curPaper);
-    if (voteType === UPVOTE) {
-      postUpvote(curPaper.id);
-    } else {
-      postDownvote(curPaper.id);
-    }
-  }
+  const onDownvote = createVoteHandler({
+    currentAuthor: currentUser?.author_profile,
+    currentVote: voteState,
+    documentCreatedBy: created_by ?? uploaded_by,
+    documentID: id,
+    documentType: nullthrows(formattedDocType, "Cannot vote without doctype"),
+    onError: emptyFncWithMsg,
+    onSuccess: ({ increment, voteType }): void => {
+      setVoteState(voteType);
+      setScore(score + increment);
+    },
+    voteType: DOWNVOTE,
+  });
+  const onUpvote = createVoteHandler({
```

Comment:
I recently refactored `createVoteHandler` into its own file because I wanted to remove as much as logic as possible from newly created `DocumentHeader` component. I also realized there is a lot of code around the handler that wasn't being used. 

I know this shouldn't necessarily full on you but I think it makes sense to deal with this now while you are at it. You will need to add `hypothesis` type to `getVoteUrl` function.

For reference, you should check out `DocumentHeader`

Reply:
NVM, I see that you are using it here. Thanks!